core/src/glue.rs

```diff
@@ -27,28 +27,39 @@ impl<T, U: GStore<T> + GStoreMut<T>> Glue<T, U> {
         }
     }
 
-    pub async fn plan<Sql: AsRef<str>>(&self, sql: Sql) -> Result<Statement> {
+    pub async fn plan<Sql: AsRef<str>>(&self, sql: Sql) -> Result<Vec<Statement>> {
         let parsed = parse(sql)?;
-        let statement = translate(&parsed[0])?;
         let storage = self.storage.as_ref().unwrap();
 
-        plan(storage, statement).await
+        let mut statements = Vec::<Statement>::new();
+        let parsed_iter = parsed.iter();
+        for parsed in parsed_iter {
+            let translated = translate(parsed)?;
+            let planned = plan(storage, translated).await?;
+            statements.push(planned);
+        }
+        Ok(statements)
     }
 
-    pub fn execute_stmt(&mut self, statement: Statement) -> Result<Payload> {
+    pub fn execute_stmt(&mut self, statement: &Statement) -> Result<Payload> {
         block_on(self.execute_stmt_async(statement))
     }
 
-    pub fn execute<Sql: AsRef<str>>(&mut self, sql: Sql) -> Result<Payload> {
-        let statement = block_on(self.plan(sql))?;
+    pub fn execute<Sql: AsRef<str>>(&mut self, sql: Sql) -> Result<Vec<Payload>> {
+        let statements = block_on(self.plan(sql))?;
 
-        self.execute_stmt(statement)
+        let mut payloads = Vec::<Payload>::new();
+        for statement in statements.iter() {
+            let payload = self.execute_stmt(statement)?;
+            payloads.push(payload);
+        }
+        Ok(payloads)
```

Comment:
```suggestion
        statements
            .iter()
            .map(|statement| self.execute_stmt(statement))
            .collect()
```
we recommend fp style code 22 🚀